motor_vehicle_report.go

```diff
@@ -0,0 +1,94 @@
+package checkr
+
+import (
+	"encoding/json"
+	"time"
+)
+
+// MotorVehicleReport ...
+// https://docs.checkr.com/#operation/getMotorVehicleReport
+type MotorVehicleReport struct {
+	ID                    string      `json:"id,omitempty"`
+	Object                string      `json:"object,omitempty"`
+	URI                   string      `json:"uri,omitempty"`
+	Status                string      `json:"status,omitempty"`
+	CreatedAt             time.Time   `json:"created_at,omitempty"`
+	CompletedAt           time.Time   `json:"completed_at,omitempty"`
+	TurnaroundTime        int         `json:"turnaround_time,omitempty"`
+	FullName              string      `json:"full_name,omitempty"`
+	LicenseNumber         string      `json:"license_number,omitempty"`
+	LicenseState          string      `json:"license_state,omitempty"`
+	PreviousLicenseNumber string      `json:"previous_license_number,omitempty"`
+	PreviousLicenseState  string      `json:"previous_license_state,omitempty"`
+	LicenseStatus         string      `json:"license_status,omitempty"`
+	LicenseType           string      `json:"license_type,omitempty"`
+	LicenseClass          string      `json:"license_class,omitempty"`
+	Dob                   string      `json:"dob,omitempty"`
+	ExpirationDate        string      `json:"expiration_date,omitempty"`
+	CovidExtension        bool        `json:"covid_extension,omitempty"`
+	IssuedDate            string      `json:"issued_date,omitempty"`
+	FirstIssuedDate       string      `json:"first_issued_date,omitempty"`
+	InferredIssuedDate    string      `json:"inferred_issued_date,omitempty"`
+	Restrictions          []string    `json:"restrictions,omitempty"`
+	CustomRules           interface{} `json:"custom_rules,omitempty"`
```

Comment:
The docs shows this is an array of strings. This is where custom Unmarshaling comes into play.

Reply:
i am not sure we have to do it. since the documentation says it is string array but we get a empty object. so we can write custom un marshall where we do each field unmarshal. since it is custom rules can't we just leave it interface{} and any way i don't we are going to use in our database.  If i had to implement what do i change the field of customRules to. If i change to array it is going to err out in unmarshal JSON. instead we can keep it as interface write unmarshal JSON and later type case custom Rules to string array. if it is valid we use it otherwise leave it.

```
func (m *MotorVehicleReport) UnmarshalJSON(b []byte) error {
	var mvr MotorVehicleReport
	err := json.Unmarshal(b, &mvr)
	if err != nil {
		return err
	}
     cr, ok := mvr.CustomRules.([]string)
     if !ok{
         return "custom rules is not string array"
      }
	return nil
}
```

Reply:
So what you can do here is create a type for `CustomRules`. 

```
func (m *MotorVehicleCustomRules) UnmarshalJSON(b []byte) error {
  // it didn't return an array but an object
  if bytes.HasPrefix(b, "{") {
    return nil
  }

  // it returned the array
  if bytes.HasPrefix(b, "[") {
      if err := json.Unmarshal(b, &m); err != nil {
	return err
      }
  }

  return nil
}
```

Reply:
@magicalbanana  although it is a good idea i am sort of thinking we are un necessarily complicating this with un marshaljson. why can't we just leave it as interface and accept what they send. what is the probability of we using that custom rules field in our code. if it is not much i kind of feel we should just leave it as interface{} and ignore that